src/RevisionPopupWidget.js

```diff
@@ -48,29 +51,52 @@ RevisionPopupWidget.prototype.show = function ( data, $target ) {
 		contribsUrl = mw.util.getUrl( `Special:Contributions/${username}` ),
 		// We typically link to Special:Contribs for IPs.
 		userPageUrl = isIP ? contribsUrl : mw.util.getUrl( `User:${data.username}` ),
-		userLinks = `
-			<a href="${userPageUrl}">${username}</a>
-			${mw.msg( 'parentheses-start' )}<a href="${mw.util.getUrl( `User talk:${username}` )}">${mw.msg( 'talkpagelinktext' )}</a>
-			${mw.msg( 'pipe-separator' )}
-			<a href="${contribsUrl}">${mw.msg( 'contribslink' )}</a>${mw.msg( 'parentheses-end' )}
-		`,
+		// Create links using the jQuery objects so they're properly escaped
+		$userLinks = $( [] )
+			.add(
+				$( '<a>' )
+					.attr( 'href', userPageUrl )
+					.append( Tools.bidiIsolate( username ) )
+			)
+			.add( document.createTextNode( ' ' + mw.msg( 'parentheses-start' ) ) )
+			.add(
+				// Talk page
+				$( '<a>' )
+					.attr( 'href', mw.util.getUrl( `User talk:${username}` ) )
+					.text( mw.msg( 'talkpagelinktext' ) )
+			)
+			.add( document.createTextNode( ' ' + mw.msg( 'pipe-separator' ) + ' ' ) )
+			.add(
+				$( '<a>' )
+					.attr( 'href', contribsUrl )
+					.text( mw.msg( 'contribslink' ) )
+			)
+			.add( document.createTextNode( mw.msg( 'parentheses-end' ) ) ),
 		dateStr = moment( data.revisionTime ).locale( mw.config.get( 'wgUserLanguage' ) ).format( 'LLL' ),
-		diffLink = `<a href="${mw.util.getUrl( `Special:Diff/${data.revisionId}` )}">${dateStr}</a>`,
-		addedMsg = mw.message( 'ext-whowrotethat-revision-added', userLinks, diffLink ).parse(),
-		commentMsg = data.comment ? `<span class="comment comment--without-parentheses ext-wwt-revisionPopupWidget-comment">${data.comment}</span>` : '',
+		// Use jQuery to make sure attributes are properly escaped
+		$diffLink = $( '<a>' )
+			.attr( 'href', mw.util.getUrl( `Special:Diff/${data.revisionId}` ) )
+			.text( dateStr ),
+		addedMsg = mw.message( 'ext-whowrotethat-revision-added', $userLinks, $diffLink ).parse(),
+		commentMsg = data.comment ?
+			`<span class="comment comment--without-parentheses ext-wwt-revisionPopupWidget-comment">${Tools.bidiIsolate( $.parseHTML( data.comment ), true )}</span>` :
```

Comment:
Do we need `$.parseHTML` here in addition to line 86?

Reply:
Probably not? the inside (which we're not sure about) is parseHTML'ed, but the outside we wrote ourselves, so it's less crucial. The bigger issue is when we can't be 100% sure about the value of what we're getting -- which is true about the comment and the username.